go-controller/pkg/ovn/pods.go

```diff
@@ -327,7 +331,8 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
 
 	if annotation == nil {
 		gatewayCIDR, _ := util.GetNodeWellKnownAddresses(nodeSubnet)
-		routes, gwIP, err := getRoutesGatewayIP(pod, gatewayCIDR)
+		namespaceAnnotations, err := oc.kube.GetAnnotationsOnNamespace(pod.Namespace)
```

Comment:
There's no guarantee that the namespace exists yet, plus you need to use the cache here (factory.go) instead of direct to the apiserver (kube.go)